### DISCUSSION TITLE
Language tags

### DISCUSSION THREAD

**Discussion by LucyOne** [koreader/koreader#8011] [Q&A]
I have books in English and German in my Calibre library. When I add a new book, I download the metadata, do an EPUB to EPUB conversion and then polish it. During the conversion, Calibre adds this line to the titlepage of the EPUB, regardless of the book language:
`<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">.`

The problem is that **xml:lang** seems to override the metadata language setting ("dc:language" in content.opf) when reading the book in KOReader. So KOReader on my Pocketbooks recognizes those books as English, not as German, and sets hyphenation accordingly. When I change it manually to **xml:lang="de"**, the books are recognized as German in KOReader.

Is there a setting to have KOReader choose the dc:language tag over the xml:lang one?

**Comment by poire-z** [comment]
I believe unchecking this should solve your use case:
![image](https://user-images.githubusercontent.com/24273478/126678039-b07227a1-8ff4-443e-845a-f8a6736d2e63.png)

and it should still use the book language from the metadata as the main language for typography.

It's wrong for calibre to set this: this could mean this HTML fragment from the book has english content.
By disable that toggle, if you have in your book a paragraph with 
`<div lang="bg"> ... ` , you won't have bulgarian hyphenation and specific glyphs enabled for this paragraph (so, these lang= tags can override the book metadata language for small sections).
But the real fix for you to not lose this nice feature is to fix this on the calibre side (or to get it fixed).

**Comment by NiLuJe** [comment]
IIRC, calibre doesn't enforce English here, it follows its own language metadata field.

**Comment by LucyOne** [comment]
It seems that this won't be fixed on Calibre's side. Kovid Goyal replied that lang attributes on individual html files must not override book language and that epubcheck complains if there is not language set for the titlepage.

**Reply by Frenzie** [reply]
Are you sure you're paraphrasing that correctly? Because (xml:)lang on an element has priority over what's specified on a parent element or in some kind of global metadata.

Spec here for reference:
https://html.spec.whatwg.org/multipage/dom.html#attr-lang

**Comment by LucyOne** [comment]
That's what he said exactly. So, he's got it wrong?

**Reply by Frenzie** [reply]
It sounds like a miscommunication of some sort but as stated there it's wrong, yes.

**Comment by LucyOne** [comment]
Here's Kovid's reply:
The lang attribute in an individual html file inside an ebook DOES NOT override the overall language for the book. Your PocketBook [he means the KOReader developers] developers dont seem to understand what an ebook is. The spec they quote is for a *single html document*, which is completely irrelevant for an ebook tht can be composed of multiple html documents each having their own potentially conflicting lang attributes.

The overall language of an ebook must be read from the metadata of the book. In the case fo EPUB books that means from the <metadata> section of the OPF file. Relevant spec: [http://idpf.org/epub/30/spec/epub30-publications.html#sec-metadata-elem](url)

**Reply by Frenzie** [reply]
Of course it doesn't override it for "the book," it overrides it for *that part of the book*.

**Reply by Frenzie** [reply]
As I said, I think there's some miscommunication at work here. But it would be preposterous to claim a book can have only one single language in it.

**Comment by LucyOne** [comment]
What happens in KOReader, though, is that the hyphenation and typography rules follow the language tags in the titlepage, not the dc:language tag in the metadata. So, in KOReader it does override it for the whole book.

**Reply by Frenzie** [reply]
Well, that's rather important context to include and it explains the miscommunication. Because *every single language attribute* very much overrides whatever its parent claims to be the case. Whether that parent happens to take the more esoteric form of EPUB metadata is irrelevant.
